### PR TITLE
Simplify PR-6334

### PR DESCRIPTION
**Pull request by sklam**
Based on #6334.

It looks like the same can be achieved without adding a `builder` argument to `get_getattr`.

### CODE REVIEW

File: numba/tests/test_objects.py [hunk 1]
```diff
@@ -30,6 +30,12 @@ def delattr_usecase(o):
 
 
 class TestAttributes(TestCase):
+    def test_getattr(self, flags=enable_pyobj_flags):
+        import numba.tests.usecases as uc
+        @njit
+        def f():
+            return uc._GLOBAL_STR
+        self.assertEquals(f(), "abc")
```
Comment (stuartarchibald) [inline numba/tests/test_objects.py:38]: Puzzled as to why this test is here? This module seems to be related to "objects" but this is compiled under `nopython` mode?

File: numba/tests/test_objects.py [hunk 2]
```diff
@@ -30,6 +30,12 @@ def delattr_usecase(o):
 
 
 class TestAttributes(TestCase):
+    def test_getattr(self, flags=enable_pyobj_flags):
```
Comment (stuartarchibald) [inline numba/tests/test_objects.py:33]: `flags` unused ?

File: numba/tests/usecases.py [hunk 3]
```diff
@@ -2,6 +2,8 @@
 import numpy as np
 from numba import jit
 
+_GLOBAL_STR = "abc"
```
Comment (stuartarchibald) [inline numba/tests/usecases.py:5]: As a total aside, this file itself seems a little strange with respect to the rest of testing, is it perhaps a legacy item?

File: numba/tests/test_objects.py (hunk 1 above)
Comment (sklam) [reply numba/tests/test_objects.py:38]: Good point. I didn't pay attention to the test file when I reviewed the original code.

Comment (sklam) [reply numba/tests/test_objects.py:38]: i think `test_auto_constants.py` is the closest test file.

Comment (stuartarchibald) [reply numba/tests/test_objects.py:38]: Looks good.

File: numba/tests/usecases.py (hunk 3 above)
Comment (sklam) [reply numba/tests/usecases.py:5]: This file along with `test_usecases.py` are probably the oldest surviving test files and they are exercising the essentials of the core compiler.

File: numba/tests/test_objects.py (hunk 2 above)
Comment (sklam) [reply numba/tests/test_objects.py]: removed

**Comment by sklam** [comment]
All comments are addressed.

File: numba/tests/test_auto_constants.py [hunk 4]
```diff
@@ -27,6 +29,12 @@ def pyfunc():
 
         self.assertEqual(pyfunc(), cfunc())
 
+    def test_module_string_constant(self):
+        @njit
+        def f():
+            return uc._GLOBAL_STR
+        self.assertEquals(f(), "abc")
```
Comment (stuartarchibald) [inline numba/tests/test_auto_constants.py:36]:
```suggestion
        self.assertEquals(f(), f.py_func())
```
saves keeping in sync

File: numba/tests/test_auto_constants.py [hunk 5]
```diff
@@ -27,7 +29,12 @@ def pyfunc():
 
         self.assertEqual(pyfunc(), cfunc())
 
+    def test_module_string_constant(self):
+        @njit
+        def f():
+            return uc._GLOBAL_STR
+        self.assertEquals(f(), f.py_func())
```
Comment (stuartarchibald) [inline numba/tests/test_auto_constants.py:36]: Also, shouldn't this be `assertEqual`? `assertEquals` is a deprecated alias https://docs.python.org/3/library/unittest.html#deprecated-aliases

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch and fixes.

**Pull request merged**